mythtv/programs/mythtranscode/replex/multiplex.c

```diff
@@ -295,7 +295,11 @@ static void writeout_video(multiplex_t *mx)
 	//estimate next pts based on bitrate of this stream and data written
 	viu->dts = uptsdiff(viu->dts + ((nlength*viu->ptsrate)>>8), 0);
 
-	write(mx->fd_out, outbuf, written);
+	if (write(mx->fd_out, outbuf, written) != written) {
+	  if (mx->error++ < 10)	/* log only first few failures */
```

Comment:
Everywhere you are increasing error count, you should do so ensuring there is no chance of wraparound in the 31 bits of the positive 32-bit integer return value.  A very large file is not out of the question here, and we could get a very large number of errors should the disk be full, for instance

Reply:
Yes, I have seen large error count in my log while testing, I think this was over 12GB of failed writes:

`
E MPEG2Replex mpeg2fix.cpp:504 (WaitBuffers) thread finished with 3576138 write errors
`

But the error count doesn't matter since in the end we bail out on any amount of errors.  The odds of it rolling over to exactly zero are very small - only zero error count will keep the truncated file.  If it rolls past zero, then we just log 10 more errors and don't log the actual much larger error count.  Maybe simpler is just boolean mx->error = 1: we either have errors or we don't.